korangar/src/loaders/animation/mod.rs

```diff
@@ -0,0 +1,474 @@
+use std::cmp::{max, min};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use cgmath::Vector2;
+use derive_new::new;
+use korangar_interface::elements::PrototypeElement;
+use num::Zero;
+use wgpu::{Device, Extent3d, Queue, TextureDescriptor, TextureDimension, TextureFormat, TextureUsages};
+
+use super::error::LoadError;
+use crate::graphics::Texture;
+use crate::loaders::{ActionLoader, Actions, AnimationState, Sprite, SpriteLoader};
+use crate::{Color, EntityType};
+
+// TODO: use cache later, the memory will be increase with this hashmap, until
+// the program is out of memory.
+#[derive(new)]
+pub struct AnimationLoader {
+    device: Arc<Device>,
+    queue: Arc<Queue>,
+    #[new(default)]
+    // The string will be type of entity
+    // 0_{body_id}_{head_id}
+    // 1_{monster_id}
+    // 2_{npc_id}
+    cache: HashMap<String, AnimationData>,

```

Comment:
As I wrote below: Better find use an enum here as the key. You can also use hashbrown's hashmap here instead of the std once, since it's much faster.

Reply:
Changed to hashbrown and used the enum as key